src/DDNSUpdate/Application/Providers/GoDaddy/Configuration/GoDaddyDNSRecordValidator.cs

```diff
@@ -0,0 +1,65 @@
+﻿using DDNSUpdate.Domain;
+using FluentValidation;
+
+namespace DDNSUpdate.Application.Providers.GoDaddy.Configuration
+{
+    public class GoDaddyDNSRecordValidator : AbstractValidator<DNSRecord>
+    {
+        private const string _propertyDisallowedMessage = "The {PropertyName} field should not be set in configuration.";
+        private const int _ttlMinimumSeconds = 600;
+
+        public static readonly string DataErrorMessage = _propertyDisallowedMessage;
+        public static readonly string FlagsErrorMessage = _propertyDisallowedMessage;
+        public static readonly string NameErrorMessage = "{PropertyName} must be set.";
+        public static readonly string PortErrorMessage = _propertyDisallowedMessage;
+        public static readonly string PriorityErrorMessage = _propertyDisallowedMessage;
+        public static readonly string TagErrorMessage = _propertyDisallowedMessage;
+        public static readonly string TTLErrorMessage = $"{{PropertyName}} must be {_ttlMinimumSeconds} seconds or greater.";
+        public static readonly string TypeErrorMessage = "Use either A or AAA for the {PropertyName}.";
+        public static readonly string WeightErrorMessage = _propertyDisallowedMessage;
+
+        public GoDaddyDNSRecordValidator()
+        {
```

Comment:
Should there be a check for the ID? Should we make sure that this value is not set?

Reply:
Done